SUSE/connect-ng
# POC for integration telemetry into SUSEConnect

**DO NOT MERGE THIS!**

## How to test this

1. Download both https://github.com/SUSE/telemetry and https://github.com/SUSE/telemetry-server (yes, you need both).
2. You can deploy the server in a variety of ways. I've just gone the bare-metal route:
  - `cd telemetry-server/server/telemetry-server`
  - `mkdir -p /tmp/telemetry/{client,server} /tmp/susetelemetry`
  - `go run . --config ../../testdata/config/localServer.yaml`
3. On another terminal:
  - `cd telemetry` (NOTE: this is the client repo)
  - `cp examples/config/telemetry.yaml /tmp/susetelemetry/config.yaml`

And now you can go on with SUSEConnect the usual way. I've done it through a container. Hence, from the source code of suseconnect-ng, go to this branch and:

```
$ docker run --network=host --rm --privileged -ti -v $(pwd):/connect -v /tmp/susetelemetry:/tmp/susetelemetry  registry.suse.com/bci/golang:1.21-openssl
```

From there go to `/connect`, build the code with `make build` and do something like `./out/suseconnect -r <regcode>`. You will see messages of telemetry data being sent.
